lib/Sema/CSDiag.cpp

```diff
@@ -161,6 +161,15 @@ void constraints::simplifyLocator(Expr *&anchor,
         continue;
       }
 
+      if (auto *subscriptExpr = dyn_cast<SubscriptExpr>(anchor)) {
+        targetAnchor = nullptr;
+        targetPath.clear();
+
+        anchor = subscriptExpr->getIndex();
+        path = path.slice(1);
+        continue;
+      }
```

Comment:
What else is this going to affect?

Reply:
From what I could tell, the majority of usages of `simplifyLocator` occur within CSDiag, and the usages outside of CSDiag appear to revolve around locators to decls, which this shouldn't affect.

So as far as I can tell, the worst-case impact would be a slight change in diagnostics (but even then, it should be a change for the better, as we're now correctly simplifying a subscript argument locator to the argument expr).

That being said, looking at it again, it appears this logic isn't necessary for this patch, so I'll remove it to minimise risk.